bare_header/device.c++

```diff
@@ -77,7 +77,7 @@ string Device::def_handle_index(const node &n) {
       });
   std::transform(instance.begin(), instance.end(), instance.begin(), toupper);
 
-  return "METAL_" + name + "_" + instance;
+  return "METAL_" + name + "_IDX_" + instance;
```

Comment:
I think this is confusing, in that for below:
/* From clock@2 */                                                                                                                                                                                                                                                                           
#define METAL_FIXED_CLOCK_2_CLOCK_FREQUENCY 72000000UL                                                                                                                                                                                                                                       
#define METAL_FIXED_CLOCK_IDX_1_CLOCK_FREQUENCY 72000000UL 
IDX_1_CLOCK is no longer associated with CLOCK_2.
Better to have either /* From clock@2 */                                                                                                                                                                                                                                                                           
#define METAL_FIXED_CLOCK_2_CLOCK_FREQUENCY 72000000UL                                                                                                                                                                                                                                       
#define METAL_FIXED_CLOCK_2_IDX_0_CLOCK_FREQUENCY 72000000UL 
#define METAL_FIXED_CLOCK_2_IDX_1_CLOCK_FREQUENCY 35000000UL 

or. simply 

#define METAL_FIXED_CLOCK_2_CLOCK_FREQUENCY 72000000UL 
 since clock is already indexed.

Reply:
The whole point of the indexed defines is that they don't have the address included. In the case of clocks it's a bit confusing because they're already "indexed" by their address, but it's messy to carve out an exception for that.

Reply:
I found a solution. If the address is too small, it's already being used as an index.